server/src/main/java/org/apache/druid/server/http/CoordinatorDynamicConfigSyncer.java

```diff
@@ -0,0 +1,221 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.server.http;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.errorprone.annotations.concurrent.GuardedBy;
+import com.google.inject.Inject;
+import org.apache.druid.client.broker.BrokerClient;
+import org.apache.druid.client.broker.BrokerClientImpl;
+import org.apache.druid.discovery.DiscoveryDruidNode;
+import org.apache.druid.discovery.DruidNodeDiscoveryProvider;
+import org.apache.druid.discovery.NodeRole;
+import org.apache.druid.guice.annotations.EscalatedGlobal;
+import org.apache.druid.guice.annotations.Json;
+import org.apache.druid.java.util.common.concurrent.Execs;
+import org.apache.druid.java.util.common.logger.Logger;
+import org.apache.druid.rpc.FixedServiceLocator;
+import org.apache.druid.rpc.ServiceClientFactory;
+import org.apache.druid.rpc.ServiceLocation;
+import org.apache.druid.rpc.StandardRetryPolicy;
+import org.apache.druid.server.DruidNode;
+import org.apache.druid.server.coordinator.CoordinatorConfigManager;
+import org.apache.druid.server.coordinator.CoordinatorDynamicConfig;
+
+import javax.annotation.Nullable;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+/**
+ * Updates all brokers with the latest coordinator dynamic config.
+ */
+public class CoordinatorDynamicConfigSyncer
+{
+  private static final Logger log = new Logger(CoordinatorDynamicConfigSyncer.class);
+
+  private final CoordinatorConfigManager configManager;
+  private final ObjectMapper jsonMapper;
+  private final DruidNodeDiscoveryProvider druidNodeDiscovery;
+
+  private final ServiceClientFactory clientFactory;
+  private final ScheduledExecutorService exec;
+  private @Nullable Future<?> syncFuture = null;
+
+  @GuardedBy("this")
+  private final Set<BrokerSyncStatus> inSyncBrokers;
+  private final AtomicReference<CoordinatorDynamicConfig> lastKnownConfig = new AtomicReference<>();
+
+  @Inject
+  public CoordinatorDynamicConfigSyncer(
+      @EscalatedGlobal final ServiceClientFactory clientFactory,
+      final CoordinatorConfigManager configManager,
+      @Json final ObjectMapper jsonMapper,
+      final DruidNodeDiscoveryProvider druidNodeDiscoveryProvider
+  )
+  {
+    this.clientFactory = clientFactory;
+    this.configManager = configManager;
+    this.jsonMapper = jsonMapper;
+    this.druidNodeDiscovery = druidNodeDiscoveryProvider;
+    this.exec = Execs.scheduledSingleThreaded("CoordinatorDynamicConfigSyncer-%d");
```

Comment:
Note for follow up: There should preferably be a lifecycle stop method which shuts down this executor, as it allows for nicer cleanup of the resources used by this class.